### PR TITLE
staking-miner: CLI flag delay solution x secs

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: utils/staking-miner/src/monitor.rs [hunk 1]
```diff
@@ -198,6 +198,7 @@ macro_rules! monitor_cmd_for { ($runtime:tt) => { paste::paste! {
 				return;
 			}
 
+
```
Comment (niklasad1) [inline utils/staking-miner/src/monitor.rs:201]:
```suggestion
```

**Comment by kianenigma** [comment]
Can you also double check that: 

1. We do read the status of the _UPDATED_ queue before submitting, 
2. and that if we observe a solution in it with the same score as what we have submitted, 
3. or if it is actually full
4. we emit a log, and we don't submit?

**Comment by niklasad1** [comment]
Yeah, we check that the solution is better than the existing ones after mining but we doesn't emit logs for that.

On other hand we don't check if the queue is full as we only submit if we are better but why should we?

**Comment by niklasad1** [comment]
> Can you also double check that: 

> 1. We do read the status of the _UPDATED_ queue before submitting, 
> 2. and that if we observe a solution in it with the same score as what we have submitted, 
> 3. or if it is actually full
> 4. we emit a log, and we don't submit? 

1. Yes, that has been done all the time but I added that we read that `current account` hasn't submitted in that round as well.
2.  Added a log if that occurs but as `if-leading stategy` is used by default nothing will be sent if the scores as equal, as we compare the scores with `ElectionScore::strict_better_better(other, episilon)`
3. If it's better we will still submit but added a log if the queue is full.
4. Yes added.

**Review by kianenigma — APPROVED** [review]

File: utils/staking-miner/src/opts.rs [hunk 2]
```diff
@@ -83,6 +83,17 @@ pub(crate) struct MonitorConfig {
 	/// `--submission-strategy "percent-better <percent>"`: submit if the submission is `n` percent better.
 	#[clap(long, parse(try_from_str), default_value = "if-leading")]
 	pub submission_strategy: SubmissionStrategy,
+
+	/// Delay in number seconds to wait until starting mining a solution.
+	///
+	/// At every block when a solution is attempted
+	/// a delay can be enforced to avoid submitting at
+	/// "same time" and risk potentially races to other miners.
```
Comment (dvdplm) [inline utils/staking-miner/src/opts.rs:91]:
```suggestion
	/// "same time" and risk potential races with other miners.
```

File: utils/staking-miner/src/opts.rs [hunk 3]
```diff
@@ -83,6 +83,17 @@ pub(crate) struct MonitorConfig {
...
+	///
+	/// This may mean if there are competing solutions you might not submit
+	/// if the scores are equal once this is enabled.
```
Comment (dvdplm) [inline utils/staking-miner/src/opts.rs:94]:
```suggestion
	/// When this is enabled and there are competing solutions, your solution might not be submitted
	/// if the scores are equal.
```

**Review by dvdplm — APPROVED** [review]

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**